src/mlang/backend_ir/bir_interpreter.ml

```diff
@@ -524,6 +524,9 @@ module Make (N : Bir_number.NumberInterface) = struct
         match evaluate_expr ctx p.mir_program data.cond_expr with
         | Number f when not (N.is_zero f) -> report_violatedcondition data ctx
         | _ -> ctx)
+    | Bir.SRuleCall _ ->
+        (* removed with [Bir.get_all_statements] below *)
```

Comment:
It would be better for debug purposes to actually interpret rules one by one rather than bundling everything in one list of statements.